airbyte-cdk/python/airbyte_cdk/sources/streams/http/error_handlers/default_backoff_strategy.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+
+from typing import Optional, Union
+
+import requests
+
+from .backoff_strategy import BackoffStrategy
+
+
+class DefaultBackoffStrategy(BackoffStrategy):
+    def __init__(
+        self,
+        max_retries: int = 5,
+        max_time: int = 60 * 10,
+    ):
+        self.max_retries = max_retries
+        self.max_time = max_time
+
+    def backoff_time(self, response_or_exception: Optional[Union[requests.Response, requests.RequestException]]) -> Optional[float]:
+        """
+        Override this method to dynamically determine backoff time e.g: by reading the X-Retry-After header.
```

Comment:
Why isn't int the default behavior in the `DefaultBackoffStrategy`?

Reply:
I don't include `X-Retry-After` header by default with the thought that if/when the default backoff_time method returns `None` it raises a `DefaultBackoffException` instead of a `UserDefinedBackoffException`.